test/async-await.js

```diff
@@ -334,6 +313,154 @@ function asyncTest (t) {
     }
   })
 
+  test('does not call reply.send() twice if 204 reponse is already sent', t => {
+    t.plan(2)
+
+    const fastify = Fastify()
+
+    fastify.get('/', async (req, reply) => {
+      reply.code(204).send()
+      reply.send = () => {
+        throw new Error('reply.send() was called twice')
+      }
+    })
+
+    fastify.inject({
+      method: 'GET',
+      url: '/'
+    }, (err, res) => {
+      t.error(err)
+      t.equal(res.statusCode, 204)
+    })
+  })
+
+  test('error is logged because promise was fulfilled with undefined', t => {
+    t.plan(3)
+
+    var fastify = null
+    var stream = split(JSON.parse)
+    try {
+      fastify = Fastify({
+        logger: {
+          stream: stream,
+          level: 'error'
+        }
+      })
+    } catch (e) {
+      t.fail()
+    }
+
+    t.tearDown(fastify.close.bind(fastify))
+
+    fastify.get('/', async (req, reply) => {
+      reply.code(200)
+    })
+
+    stream.once('data', line => {
+      t.ok(line.msg, 'Promise may not be fulfilled with \'undefined\' when statusCode is not 204')
```

Comment:
I may be wrong, but this check should be `t.strictEqual` or `t.ok(line.msg === '...')`
https://www.node-tap.org/asserts/

Reply:
Good catch!